qa/backwards-5.0/src/test/java/org/elasticsearch/backwards/IndexingIT.java

```diff
@@ -94,6 +91,67 @@ protected int indexDocs(String index, final int idStart, final int numDocs) thro
         return numDocs;
     }
 
+    private void indexDocs(String index, int numDocs) throws IOException {
+        for (int i = 0; i < numDocs; i++) {
+            assertOK(client().performRequest("POST", index + "/test", emptyMap(),
+                    new StringEntity("{\"test\": \"test_" + i + "\"}", ContentType.APPLICATION_JSON)));
+        }
+    }
+
+    public void testIndexVersionPropagation() throws Exception {
+        Nodes nodes = buildNodeAndVersions();
+        assumeFalse("new nodes is empty", nodes.getNewNodes().isEmpty());
+        logger.info("cluster discovered: {}", nodes.toString());
+        final List<String> bwcNamesList = nodes.getBWCNodes().stream().map(Node::getNodeName).collect(Collectors.toList());
+        final String bwcNames = bwcNamesList.stream().collect(Collectors.joining(","));
+        Settings.Builder settings = Settings.builder()
+                .put(IndexMetaData.INDEX_NUMBER_OF_SHARDS_SETTING.getKey(), 1)
+                .put(IndexMetaData.INDEX_NUMBER_OF_REPLICAS_SETTING.getKey(), 2)
+                .put("index.routing.allocation.include._name", bwcNames);
+
+        final boolean checkGlobalCheckpoints = nodes.getMaster().getVersion().onOrAfter(Version.V_6_0_0_alpha1_UNRELEASED);
+        logger.info("master version is [{}], global checkpoints will be [{}]", nodes.getMaster().getVersion(),
```

Comment:
I don't think we need to check checkpoints in that test? it just adds noise, no?